src/util.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Developed by Claudio André <claudioandre.br at gmail.com> in 2020
+ *
+ * Copyright (c) 2020 Claudio André <claudioandre.br at gmail.com>
+ *
+ * This program comes with ABSOLUTELY NO WARRANTY; express or implied .
+ * This is free software, and you are welcome to redistribute it
+ * under certain conditions; as expressed here
+ * http://www.gnu.org/licenses/gpl-2.0.html
+ */
+
+#ifndef __MINGW32__
+#include <sys/ioctl.h>
+#endif
+#include <stdarg.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include "config.h"
+#include "john.h"
+#include "logger.h"
+#include "options.h"
+#include "util.h"
+
+static int is_separator(char symbol)
+{
+
+	if (symbol == ' ' || symbol == '\t' || symbol == '\n' || symbol == '\0')
+		return 1;
+
+	return 0;
+}
+
+static inline int wordlen(const char * str)
+{
+	int len = 0;
+
+	while (!is_separator(str[len]))
+		len++;
+
+	return len;
+}
+
+static int wrap(const char * text, char * line, const int width, int first_line,
+    int identation)
+{
+	int line_len = 0, start = 0, limit;
+
+	/* ltrim text. */
+	while (!first_line && text[start] != '\0' && is_separator(text[start]))
+		start++;
+
+	/* Protection against invalid data. */
+	if (identation > width || first_line)
+		identation = 0;
+	limit = width - identation;
+
+	while (text[line_len] != '\0') {
+
+		if (is_separator(text[line_len]))
+			if ((text[line_len] == '\n') ||
+			     line_len + wordlen(&text[line_len + 1]) >= limit)
+				break;
+
+		line_len++;
+
+		/* Separator was not found. */
+		if (line_len == limit)
+			break;
+	}
+
+	if (identation)
+		memset(line, ' ', identation);
+	strncpy(line + identation, text + start, limit);
+	line[line_len + identation] = '\0';
+
+	return line_len;
+}
+
+int get_windows_size(int *lines, int *cols)
+{
+#ifndef __MINGW32__
+	struct winsize w;
+
+	if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &w) == 0) {
+		const char *tmp_value;
+		int limit = 0;
+
+		*(lines) = w.ws_row;
+		*(cols) = w.ws_col;
+
+		if ((tmp_value = cfg_get_param(SECTION_OPTIONS, "", "MaxConsoleCols")))
+			limit = atoi(tmp_value);
+
+		/* Never use wider than n or john.conf option. */
+		if (limit && *(cols) > limit)
+			*(cols) = limit;
+		if (*(cols) > 140)
+			*(cols) = 140;
+		if (*(cols) < 60)
+			*(cols) = 60;
+		return 0;
+	}
+#endif
+	*(lines) = 20;
```

Comment:
I think default height would traditionally be 24.

Reply:
I already changed this (to 24x80). The default in my (GNOME) terminal.